### PR TITLE
feat: file upload endpoint that allows searching database

### PR DESCRIPTION
**Pull request by pelavarthi**
# Pull request

**Resolves:** #<<66>>

### CODE REVIEW

**Review by Viraj-Ganguli — COMMENTED** [review]
Might be more efficient to import the getHash function but otherwise looks good!

**Review by DoodlePooge — CHANGES_REQUESTED** [review]
I would like to see some sort of response message with a proper status code if there are no matches found. This should include a message and the data of the file that was uploaded.

**Comment by XxXgamerguyXxX** [comment]
Looks good

File: pyproject.toml [hunk 1]
```diff
@@ -24,6 +24,7 @@ beanie = "^1.19.2"
 typer = { extras = ["all"], version = "^0.9.0" }
 jinja2 = "^3.1.2"
 pytest = "^7.4.0"
+python-multipart = "^0.0.6"
```
Comment (DoodlePooge) [inline pyproject.toml:27]: what is this used for? It looks like a library for testing which again should not be a production dependency. If we do need this add it by `poetry add [package] --group dev`

File: darkmoon/api/v1/metadata/schema.py [hunk 2]
```diff
@@ -169,3 +169,11 @@ class UploadResponse(Response):
     data: Metadata = Field(
         description="The object inserted or updated on the database",
     )
+
+
+class UploadResponse2(Response):
```
[the remark this answers is not in the document]
Comment (DoodlePooge) [reply darkmoon/api/v1/metadata/schema.py:174]: In fact rename both so they make sense for their use cases, it's not a global UploadResponse if it has Metadata as its data

**Review by DoodlePooge — DISMISSED** [review]

File: pyproject.toml (hunk 1 above)
Comment (XxXgamerguyXxX) [reply pyproject.toml:27]: We're going to use it and we used the command you suggested. We need it to upload files.

**Comment by Synthint** [comment]
unused test_metadata_entity in test_post_hash_comparison

Both testHashComparisonFile and testHashComparisonWithoutFile are in camel case, reformat so it is in snake case where there are underscores between words and no capitalization.

**Comment by asmitakarandikar** [comment]
Remove all of the unnecessary comments in views.py

File: tests/integration/api/Test Name
Comment (tclasen) [inline tests/integration/api/Test Name:1]: Please remove this file.

File: darkmoon/api/v1/metadata/views.py [hunk 3]
```diff
@@ -287,3 +300,104 @@ async def upload_metadata(
...
+                {
+                    "file_type": inputFileType,
+                },
+            ],
+        }
+
+        results = await collection.find(search_query).to_list(length=length)
+        li = []
+        for item in results:
+            li.append(MetadataEntity.parse_obj(item))
```
Comment (tclasen) [inline darkmoon/api/v1/metadata/views.py:373]: Please refactor this into a list comprehension.

File: tests/integration/api/Test Name
Comment (XxXgamerguyXxX) [reply tests/integration/api/Test Name:1]: We are using this file for testing for the test_post_hash_comparision method in test_api.py.

Comment (tclasen) [inline tests/integration/api/Test Name:1]: Please delete this file.

**Review by tclasen — CHANGES_REQUESTED** [review]

**Review by tclasen — APPROVED** [review]

**Pull request merged**